src/Symfony/Component/Form/FormRegistry.php

```diff
@@ -106,18 +109,33 @@ private function resolveType(FormTypeInterface $type)
         $parentType = $type->getParent();
         $fqcn = get_class($type);
 
-        foreach ($this->extensions as $extension) {
-            $typeExtensions = array_merge(
+        if (isset($this->checkedTypes[$fqcn])) {
+            $types = implode(' > ', array_merge(array_keys($this->checkedTypes), array($fqcn)));
+            throw new LogicException(sprintf('Circular reference detected for form "%s" (%s).', $fqcn, $types));
+        }
+
+        $this->checkedTypes[$fqcn] = true;
+
+        try {
+            if ($parentType === $fqcn) {
```

Comment:
one last thing sorry: this if can be moved before L117, isn't it?

Reply:
I'm wondering if this is even necessary? Without it, you would still get the circular ref error. Or is it better to be explicit in checking the same type as the parent?

Reply:
let's remove it then you're right